Sources/Hummingbird/Configuration.swift

```diff
@@ -18,240 +18,238 @@ import Logging
+        maxUploadSize: Int = 1 * 1024 * 1024,
+        maxStreamedUploadSize: Int = 4 * 1024 * 1024,
+        maxStreamingBufferSize: Int = 1 * 1024 * 1024,
+        reuseAddress: Bool = true,
+        enableHttpPipelining: Bool = true,
+        idleTimeoutConfiguration: IdleStateHandlerConfiguration? = nil,
+        threadPoolSize: Int = 2,
+        logLevel: Logger.Level? = nil,
+        noHTTPServer: Bool = false,
+        tlsOptions: TSTLSOptions
+    ) {
+        let env = HBEnvironment()
+
+        self.address = address
+        self.serverName = serverName
+        self.maxUploadSize = maxUploadSize
+        self.maxStreamedUploadSize = maxStreamedUploadSize
+        self.maxStreamingBufferSize = maxStreamingBufferSize
+        self.backlog = 256 // not used by Network framework
+        self.reuseAddress = reuseAddress
+        self.tcpNoDelay = true // not used by Network framework
+        self.enableHttpPipelining = enableHttpPipelining
+        self.idleTimeoutConfiguration = idleTimeoutConfiguration
+        self.tlsOptions = tlsOptions
+
+        self.threadPoolSize = threadPoolSize
+        self.noHTTPServer = noHTTPServer
+
+        if let logLevel = logLevel {
+            self.logLevel = logLevel
+        } else if let logLevel = env.get("LOG_LEVEL") {
+            self.logLevel = Logger.Level(rawValue: logLevel) ?? .info
+        } else {
+            self.logLevel = .info
         }
+    }
 
-        #endif
-
-        /// Create new configuration struct with updated values
-        public func with(
-            address: HBBindAddress? = nil,
-            serverName: String? = nil,
-            maxUploadSize: Int? = nil,
-            maxStreamingBufferSize: Int? = nil,
-            backlog: Int? = nil,
-            reuseAddress: Bool? = nil,
-            tcpNoDelay: Bool? = nil,
-            enableHttpPipelining: Bool? = nil,
-            idleTimeoutConfiguration: IdleStateHandlerConfiguration? = nil,
-            threadPoolSize: Int? = nil,
-            logLevel: Logger.Level? = nil
-        ) -> Self {
-            return Configuration(
-                address: address ?? self.address,
-                serverName: serverName ?? self.serverName,
-                maxUploadSize: maxUploadSize ?? self.maxUploadSize,
-                maxStreamingBufferSize: maxStreamingBufferSize ?? self.maxStreamingBufferSize,
-                backlog: backlog ?? self.backlog,
-                reuseAddress: reuseAddress ?? self.reuseAddress,
-                tcpNoDelay: tcpNoDelay ?? self.tcpNoDelay,
-                enableHttpPipelining: enableHttpPipelining ?? self.enableHttpPipelining,
-                idleTimeoutConfiguration: idleTimeoutConfiguration ?? self.idleTimeoutConfiguration,
-                threadPoolSize: threadPoolSize ?? self.threadPoolSize,
-                logLevel: logLevel ?? self.logLevel
-            )
-        }
+    #endif
+
+    /// Create new configuration struct with updated values
+    public func with(
```

Comment:
`with` is a bit of an unclear name. Can we rename it to `modifying`? I don't have a good name at hand, just feel that `with` is a bit unclear.

Reply:
Not sure of an alternative but this is slightly outside of the confines of this PR.